lms/djangoapps/verify_student/management/commands/populate_expiry_date.py

```diff
@@ -0,0 +1,104 @@
+"""
+Django admin command to populate expiry_date for approved verifications in SoftwareSecurePhotoVerification
+"""
+import logging
+import time
+from datetime import timedelta
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.db.models import F
+from util.query import use_read_replica_if_available
+
+from lms.djangoapps.verify_student.models import SoftwareSecurePhotoVerification
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    This command sets the expiry_date for users for which the verification is approved
+
+    The task is performed in batches with maximum number of rows to process given in argument `batch_size`
+    and a sleep time between each batch given by `sleep_time`
+
+    Default values:
+        `batch_size` = 1000 rows
+        `sleep_time` = 10 seconds
+
+    Example usage:
+        $ ./manage.py lms populate_expiry_date --batch_size=1000 --sleep_time=5
+    OR
+        $ ./manage.py lms populate_expiry_date
+    """
+    help = 'Populate expiry_date for approved verifications'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--batch_size',
+            action='store',
+            dest='batch_size',
+            type=int,
+            default=1000,
+            help='Maximum number of database rows to process. '
+                 'This helps avoid locking the database while updating large amount of data.')
+        parser.add_argument(
+            '--sleep_time',
+            action='store',
+            dest='sleep_time',
+            type=int,
+            default=10,
+            help='Sleep time in seconds between update of batches')
+
+    def handle(self, *args, **options):
+        """
+        Handler for the command
+
+        It filters approved Software Secure Photo Verification and then for each distinct user it finds the most
+        recent approved verification and set its expiry_date
+        """
+        batch_size = options['batch_size']
+        sleep_time = options['sleep_time']
+
+        query = SoftwareSecurePhotoVerification.objects.filter(status='approved').order_by()
+        sspv = use_read_replica_if_available(query)
+
+        if not sspv.count():
+            logger.info("No approved entries found in SoftwareSecurePhotoVerification")
+            return
+
+        distinct_user_ids = []
+        update_verification_ids = []
```

Comment:
Same for this, you probably want a set here so you can guarantee uniqueness without having to check it every time.

Reply:
This is used to just batch the ids. No uniqueness check is performed on it as the condition `if verification.user_id not in distinct_user_ids:` already makes sure only unique ids get through

Reply:
Makes Sense.